src/frontend/src/lib/utils/credentials.utils.ts

```diff
@@ -0,0 +1,22 @@
+import type { UserCredential, UserProfile } from '$declarations/backend/backend.did';
+import { POUH_CREDENTIAL_TYPE } from '$lib/constants/credentials.constants';
+import { isNullish } from '@dfinity/utils';
+
+const isValidPouhCredential = (credential: UserCredential): boolean =>
+	POUH_CREDENTIAL_TYPE in credential.credential_type &&
+	credential.verified_date_timestamp.length > 0;
```

Comment:
What I meant it that `credential.verified_date_timestamp.length > 0` is the criterum that tells if a credential is valid or not.

`POUH_CREDENTIAL_TYPE in credential.credential_type` tells if there is a credential of specific type or none of that type.

So I would have done:

```
const isValidPouhCredential = (credential: UserCredential): boolean =>
	credential.verified_date_timestamp.length > 0;

export const hasPouhCredential = (profile: UserProfile | null | undefined): boolean | undefined => {
	if (isNullish(profile)) {
		return undefined;
	}
	return profile.credentials.some((credentials) => POUH_CREDENTIAL_TYPE in credential.credential_type && isValidPouhCredential(credential));
};
```

BUT, it's just maybe me that does not understand how it works out so code wise it's all good so I approve the PR.

Reply:
I refactored a bit more. I hope it's clearer now. I understand the confusion.

Reply:
Yes, thanks!